guide/interactions/buttons.md

```diff
@@ -0,0 +1,198 @@
+	<discord-message profile="bot">
+		Pong! (ephemeral + components)
+		<discord-embed
+			slot="embeds"
+			color="#0099ff"
+			title="Some title"
+			url="https://discord.js.org/"
+		>
+			Some description here
+		</discord-embed>
+	</discord-message>
+</div>
+
+Now you know all there is to building and sending a `MessageButton`! Let's move on to how to receive buttons!
+
+
+## Receiving buttons
+
+Whilst you can receive and handle a `ButtonInteraction` via the interaction event, we reccomend you use one of the collectors we'll be covering in the next section. Now to receive a `ButtonInteraction`, simply attach an event listener to your client and also use the `Interaction#isButton()` typeguard to make sure you only receive button interactions:
+
+```js {2}
+client.on('interaction', interaction => {
+	if (!interaction.isButton()) return;
+	console.log(interaction);
+});
+```
+
+
+## Button collectors
+
+These work quite similarly to message collectors, except that you apply them on a message rather than a channel.
+
+### ButtonInteractionCollector
+
+To create a basic event-based `ButtonInteractionCollector` simply do as follows:
+
+```js
+client.on('message', message => {
+	const filter = interaction => interaction.customID === 'primary' && interaction.user.id === '122157285790187530';
+
+	const collector = message.createButtonInteractionCollector(filter, { time: 15000 });
+	collector.on('collect', i => console.log(`Collected ${i.customID}`));
+	collector.on('end', collected => console.log(`Collected ${collected.size} items`));
+});
+```
+
+### awaitButtonInteractions
+
+As with other types of collectors, you can also use a promise-based collector like this:
+
+```js {4-6}
+client.on('message', message => {
+	const filter = interaction => interaction.customID === 'primary' && interaction.user.id === '122157285790187530';
+
+	message.awaitButtonInteraction(filter, { time: 15000 })
+		.then(collected => console.log(`Collected ${collected.size} interactions`))
+		.catch(console.error);
+});
+```
+
+
+## Responding to buttons
+
+The `ButtonInteraction` class has similar methods as the `CommandInteraction` class, which we we'll be covering in the following section:
+
+::: tip
+The following methods behave exactly the same as on the `CommandInteraction` class:
+- `reply()`
+- `editReply()`
+- `defer()`
+- `fetchReply()`
+- `deleteReply()`
+- `followUp()`
+:::
+
+### Updating the button message
+
+The `ButtonInteraction` class provides a method to update the message the button is attached to, by using `ButtonInteraction#update()` as follows:
+
+```js {1,3}
+client.on('interaction', async interaction => {
+	if (!interaction.isButton()) return;
+	if (interaction.customID === 'primary') await interaction.update('A button was clicked', { components: [] });
+});
+```
+
+### Deferring and updating the button message
+
+Additionally to deferring the response of the interaction, you can defer the button, which will trigger a loading state:
```

Comment:
Yes, I remember our conversation on it, I just did decide since this is still a draft I will preliminary just go with what the official description says until Discord finally sorts out their stuff as you said

Reply:
likely won't wait for the official documentation to release before merging this into v13-prep, so we can move on with porting the rest of the guide code ad code samples ahead of time

if you are unsure decide for one and create an issue, so we can keep track of it later on, best with a reference to the section of course